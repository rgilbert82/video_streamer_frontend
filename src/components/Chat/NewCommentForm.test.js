import React              from 'react';
import { shallow }        from 'enzyme';
import { NewCommentForm } from '.';

describe(NewCommentForm, () => {
  const createMock = jest.fn();
  let component;

  beforeEach(() => {
    component = shallow(<NewCommentForm displayMessage={jest.fn()} createNewComment={createMock} />);
  });

  it('renders properly', () => {
    expect(component).toMatchSnapshot();
  });

  it("updates the body", () => {
    expect(component.state('body')).toEqual('');
    component.find('#form_body').simulate('change', { target: { value: 'Hello World' }});
    expect(component.state('body')).toEqual('Hello World');
  });

  it("is a valid form", () => {
    expect(component.instance().validForm()).toEqual(false);
    component.find('#form_body').simulate('change', { target: { value: 'Hello World' }});
    expect(component.instance().validForm()).toEqual(true);
  });

  it("submits the form", () => {
    component.find('#form_body').simulate('change', { target: { value: 'Hello World' }});
    component.find('#submit_comment_form').simulate('click');
    expect(createMock).toBeCalled();
  });
});
