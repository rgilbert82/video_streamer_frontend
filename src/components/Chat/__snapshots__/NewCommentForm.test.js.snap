// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`class NewCommentForm extends _react2.default.Component {
  constructor(props) {
    super(props);

    this.state = {
      body: '' };


    this.updateBody = this.updateBody.bind(this);
    this.createComment = this.createComment.bind(this);
    this.validForm = this.validForm.bind(this);
  }

  updateBody(e) {
    this.setState({ body: e.target.value });
  }

  createComment() {
    this.props.createNewComment(this.state.body);
    this.setState({ body: '' });
  }

  validForm() {
    return this.state.body.length > 0;
  }

  render() {
    return (
      _react2.default.createElement('div', { id: 'newCommentForm', __source: { fileName: _jsxFileName, lineNumber: 32 }, __self: this },
        _react2.default.createElement(_reactBootstrap.FormGroup, { __source: { fileName: _jsxFileName, lineNumber: 33 }, __self: this },
          _react2.default.createElement(_reactBootstrap.FormControl, {
            componentClass: 'textarea',
            id: 'form_body',
            onChange: this.updateBody,
            value: this.state.body,
            placeholder: 'Add a comment', __source: { fileName: _jsxFileName, lineNumber: 34 }, __self: this })),


        _react2.default.createElement(_reactBootstrap.Button, {
            id: 'submit_comment_form',
            disabled: !this.validForm(),
            onClick: this.createComment, __source: { fileName: _jsxFileName, lineNumber: 42 }, __self: this }, 'Submit')));




  }} renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewCommentForm
    createNewComment={[Function]}
    displayMessage={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <FormGroup
          bsClass="form-group"
>
          <FormControl
                    bsClass="form-control"
                    componentClass="textarea"
                    id="form_body"
                    onChange={[Function]}
                    placeholder="Add a comment"
                    value=""
          />
</FormGroup>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={true}
          id="submit_comment_form"
          onClick={[Function]}
>
          Submit
</Button>,
      ],
      "id": "newCommentForm",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form-group",
          "children": <FormControl
            bsClass="form-control"
            componentClass="textarea"
            id="form_body"
            onChange={[Function]}
            placeholder="Add a comment"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-control",
            "componentClass": "textarea",
            "id": "form_body",
            "onChange": [Function],
            "placeholder": "Add a comment",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Submit",
          "disabled": true,
          "id": "submit_comment_form",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Submit",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormGroup
            bsClass="form-group"
>
            <FormControl
                        bsClass="form-control"
                        componentClass="textarea"
                        id="form_body"
                        onChange={[Function]}
                        placeholder="Add a comment"
                        value=""
            />
</FormGroup>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={true}
            id="submit_comment_form"
            onClick={[Function]}
>
            Submit
</Button>,
        ],
        "id": "newCommentForm",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": <FormControl
              bsClass="form-control"
              componentClass="textarea"
              id="form_body"
              onChange={[Function]}
              placeholder="Add a comment"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control",
              "componentClass": "textarea",
              "id": "form_body",
              "onChange": [Function],
              "placeholder": "Add a comment",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Submit",
            "disabled": true,
            "id": "submit_comment_form",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Submit",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
